themes/qgis/assets/js/navbar.js

```diff
@@ -1,22 +1,163 @@
+      if (result.length > 0) {
+        populateResults(result);
+      } else {
+        document.getElementById('search-results').innerHTML = '<p class=\"search-results-empty\">No matches found</p>';
+      }
+      hide(document.querySelector('.search-loading'));
+    })
+      .catch(function (err) {
+        console.log('Fetch Error :-S', err);
       });
+  });
+}
+
+function populateResults(results) {
+
+  var searchQuery = document.getElementById("search-query").value;
+  var searchResults = document.getElementById("search-results");
+
+  // pull template from hugo template definition
+  var templateDefinition = document.getElementById("search-result-template").innerHTML;
+
+  results.forEach(function (value, key) {
+
+    var contents = value.item.contents;
+    var snippet = "";
+    var snippetHighlights = [];
+
+    snippetHighlights.push(searchQuery);
+    snippet = contents.substring(0, summaryInclude * 2) + '&hellip;';
+
+    //replace values
+    var tags = ""
+    if (value.item.tags) {
+      value.item.tags.forEach(function (element) {
+        tags = tags + "<a href='/tags/" + element + "'>" + "#" + element + "</a> "
+      });
+    }
+
+    var output = render(templateDefinition, {
+      key: key,
+      title: value.item.title,
+      link: value.item.permalink,
+      tags: tags,
+      categories: value.item.categories,
+      snippet: snippet
     });
-  
-  });
\ No newline at end of file
+    searchResults.innerHTML += output;
+
+    snippetHighlights.forEach(function (snipvalue, snipkey) {
+      var instance = new Mark(document.getElementById('summary-' + key));
+      instance.mark(snipvalue);
+    });
+
+  });
+}
+
+
+// search js 
+
+function render(templateString, data) {
+  var conditionalMatches, conditionalPattern, copy;
+  conditionalPattern = /\$\{\s*isset ([a-zA-Z]*) \s*\}(.*)\$\{\s*end\s*}/g;
+  //since loop below depends on re.lastInxdex, we use a copy to capture any manipulations whilst inside the loop
+  copy = templateString;
+  while ((conditionalMatches = conditionalPattern.exec(templateString)) !== null) {
+    if (data[conditionalMatches[1]]) {
+      //valid key, remove conditionals, leave contents.
+      copy = copy.replace(conditionalMatches[0], conditionalMatches[2]);
+    } else {
+      //not valid, remove entire section
+      copy = copy.replace(conditionalMatches[0], '');
+    }
+  }
+  templateString = copy;
+  //now any conditionals removed we can do simple substitution
+  var key, find, re;
+  for (key in data) {
+    find = '\\$\\{\\s*' + key + '\\s*\\}';
+    re = new RegExp(find, 'g');
+    templateString = templateString.replace(re, data[key]);
+  }
+  return templateString;
+}
+
+// Helper Functions
+function show(elem) {
+  elem.style.display = 'block';
+}
+function hide(elem) {
+  elem.style.display = 'none';
+}
+function param(name) {
+  return decodeURIComponent((location.search.split(name + '=')[1] || '').split('&')[0]).replace(/\+/g, ' ');
+}
```

Comment:
You had all the code from search.js duplicated in here, remove and leave this file unchanged please.